src/lib/seam/components/AccessCodeForm/AccessCodeFormContent.tsx

```diff
@@ -0,0 +1,234 @@
+import classNames from 'classnames'
+import { useState } from 'react'
+import type { AccessCode } from 'seamapi'
+
+import {
+  get24HoursLater,
+  getBrowserTimezone,
+  getNow,
+  getTimezoneFromIsoDate,
+} from 'lib/dates.js'
+import type { UseAccessCodeData } from 'lib/seam/access-codes/use-access-code.js'
+import { useSubmitAdd } from 'lib/seam/components/AccessCodeForm/AccessCodeAddForm.js'
+import { useSubmitUpdate } from 'lib/seam/components/AccessCodeForm/AccessCodeEditForm.js'
+import type { AccessCodeFormProps } from 'lib/seam/components/AccessCodeForm/AccessCodeForm.js'
+import { AccessCodeFormDatePicker } from 'lib/seam/components/AccessCodeForm/AccessCodeFormDatePicker.js'
+import { AccessCodeFormTimes } from 'lib/seam/components/AccessCodeForm/AccessCodeFormTimes.js'
+import { AccessCodeFormTimezonePicker } from 'lib/seam/components/AccessCodeForm/AccessCodeFormTimezonePicker.js'
+import type { UseDeviceData } from 'lib/seam/devices/use-device.js'
+import { Button } from 'lib/ui/Button.js'
+import { FormField } from 'lib/ui/FormField.js'
+import { InputLabel } from 'lib/ui/InputLabel.js'
+import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
+import { RadioField } from 'lib/ui/RadioField/RadioField.js'
+import { TextField } from 'lib/ui/TextField/TextField.js'
+import { useToggle } from 'lib/ui/use-toggle.js'
+
+interface AccessCodeFormContentProps extends AccessCodeFormProps {
+  accessCode?: NonNullable<UseAccessCodeData>
+  device: NonNullable<UseDeviceData>
+}
+
+export default function AccessCodeFormContent({
+  className,
+  ...otherProps
+}: AccessCodeFormContentProps): JSX.Element | null {
+  return (
+    <div className={classNames('seam-access-code-form', className)}>
+      <Content {...otherProps} />
+    </div>
+  )
+}
+
+function Content({
+  onBack,
+  accessCode,
+  device,
+}: Omit<AccessCodeFormContentProps, 'className'>): JSX.Element {
+  const [name, setName] = useState(accessCode?.name ?? '')
+  const [type, setType] = useState<AccessCode['type']>(
+    accessCode?.type ?? 'ongoing'
+  )
+  const [datePickerVisible, setDatePickerVisible] = useState(false)
+  const [timezone, setTimezone] = useState<string>(
+    getAccessCodeTimezone(accessCode) ?? getBrowserTimezone()
+  )
+  const [startDate, setStartDate] = useState<string>(
+    getAccessCodeDate('starts_at', accessCode) ?? getNow()
+  )
+  const [endDate, setEndDate] = useState<string>(
+    getAccessCodeDate('ends_at', accessCode) ?? get24HoursLater()
+  )
+  const [timezonePickerVisible, toggleTimezonePicker] = useToggle()
+  const { submit: submitAdd, isLoading: isLoadingAdd } = useSubmitAdd()
+  const { submit: submitEdit, isLoading: isLoadingEdit } = useSubmitUpdate()
+  const isLoading = isLoadingAdd || isLoadingEdit
+
+  const submit = (): void => {
+    if (isLoading) {
+      return
+    }
+
+    if (accessCode != null) {
```

Comment:
I think we can simply things by passing in `submit` and `isLoading` (or maybe `isSubmitting`), but it makes the types a bit harder to work out. 🤔

Reply:
Yep, updated in https://github.com/seamapi/react/commit/140a430f4424af96a7663ef3af2bc6edf841c9cb for the `<AccessCodeForm/>` to just take an `onSubmit`. Much cleaner now IMO.